### PR TITLE
Foundational changes for Cypher chunking

### PR DESCRIPTION
**Pull request by petraselmer**
This must be merged simultaneously with https://github.com/neo-technology/neo4j-manual-modeling/pull/86

### CODE REVIEW

File: manual/cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/FunctionsTest.scala [hunk 1]
```diff
@@ -122,7 +122,7 @@ class FunctionsTest extends DocumentingTestBase {
 
   @Test def size() {
     testThis(
-      title = "size()",
+      title = "size() - list",
```
Comment (jjaderberg) [inline manual/cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/FunctionsTest.scala]: I believe the title controls the file name. With a title of "size() - list" the file name becomes "size---list.asciidoc", which means the include directive for this content fails. If you run a `find manual/cypher/cypher-docs/ -name "*size*"` you should see the generated files. Mutatis mutandis also the other "size" and the two "length" items.

File: manual/cypher/cypher-docs/src/docs/dev/clause.asciidoc [hunk 2]
```diff
@@ -0,0 +1,157 @@
+[[query-clause]]
+= Clauses
+
+[discrete]
```
Comment (jjaderberg) [inline manual/cypher/cypher-docs/src/docs/dev/clause.asciidoc]: I don't think the AsciiDoc processor knows `[discrete]` so it chokes. Can we make do without it?

**Review by jjaderberg — CHANGES_REQUESTED** [review]
Very nice!

Two trifles indicated need fixing:
- no `[discrete]` styles in AsciiDoc
- file names for include directives must agree with what is generated

There is a third after that which at a glance looks like a renegade null character, but I haven't quite figured it out yet.

**Comment by petraselmer** [comment]
@jjaderberg thanks for the comments. I have addressed the two issues you raised.

**Review by jjaderberg — APPROVED** [review]

**Pull request merged**